include/mbedtls/psa_util.h

```diff
@@ -123,9 +125,10 @@ static inline psa_key_usage_t mbedtls_psa_translate_cipher_operation(
 /* Translations for hashing. */
 
 /* Note: this function should not be used from inside the library, use
- * mbedtls_hash_info_psa_from_md() from the internal hash_info.h instead.
+ * mbedtls_md_psa_alg_from_type() from the internal md_psa.h instead.
  * It is kept only for compatibility in case applications were using it. */
-static inline psa_algorithm_t mbedtls_psa_translate_md(mbedtls_md_type_t md_alg)
+#if !defined(MBEDTLS_DEPRECATED_REMOVED)
+static inline MBEDTLS_DEPRECATED psa_algorithm_t mbedtls_psa_translate_md(mbedtls_md_type_t md_alg)
```

Comment:
This whole module is documented as “not part of the public API and may change at any time”. So we could remove the function now if we wanted. It's nice to give users a transition period, but if we're doing that, shouldn't we announce it in a changelog entry?

Reply:
I'd accept any of:

* This was a private function. We just remove it. No changelog entry.
* This was an officially private, technically public function. We deprecate it and will remove it in a future minor release. Changelog entries.
* This was an accidentally public function. We deprecate it but keep it around until the next major release. Changelog entries.

I do wonder though: when deprecating an interface but keeping the underlying feature, there's normally an alternative interface. But here we aren't offering an alternative. So shouldn't the new function be public? Sure, it wasn't a public interface, but it was part of the interface between crypto and TLS, and that's typically something that implementers of other protocols might need.

Reply:
> Sure, it wasn't a public interface, but it was part of the interface between crypto and TLS, and that's typically something that implementers of other protocols might need.

Generally speaking I agree that when an interface is needed to implement TLS, it's likely needed by other people as well. However in that specific case, the reason it's needed by our implementation TLS is that we're kind of in the middle of a transition from legacy to PSA, where all structures still store algorithms as `mbedtls_md_type_t` but (with `USE_PSA`) computations are done with PSA. I'm not sure if implementers of other protocols are likely to find themselves in that situation. Wdyt?

Reply:
I'm opting for the first option: "This was a private function. We just remove it. No changelog entry."